internal/controllers/lvmcluster/status.go

```diff
@@ -0,0 +1,161 @@
+package lvmcluster
+
+import (
+	"fmt"
+
+	lvmv1alpha1 "github.com/openshift/lvm-operator/api/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ReasonReconciliationInProgress  = "ReconciliationInProgress"
+	MessageReconciliationInProgress = "Reconciliation is still in progress"
+
+	ReasonResourcesSynced  = "ResourcesSynced"
+	MessageResourcesSynced = "Reconciliation is successfully complete"
+
+	ReasonResourcesSyncFailed        = "ResourcesSyncFailed"
+	MessageReasonResourcesSyncFailed = "Reconciliation is failed with %v"
+
+	ReasonVGReadinessInProgress  = "VGReadinessInProgress"
+	MessageVGReadinessInProgress = "VG readiness check is still in progress"
+
+	ReasonVGsFailed  = "VGsFailed"
+	MessageVGsFailed = "One or more vgs are failed"
+
+	ReasonVGsDegraded  = "VGsDegraded"
+	MessageVGsDegraded = "One or more vgs are degraded"
+
+	ReasonVGsReady  = "VGsReady"
+	MessageVGsReady = "All the VGs are ready"
+)
+
+func computeInitialConditions() []lvmv1alpha1.Condition {
+	var conditions []lvmv1alpha1.Condition
+	reconciliationCondition := computeResourcesSyncedCondition(true, nil)
+	conditions = append(conditions, reconciliationCondition)
+	vgReadinessCondition := computeVGReadinessCondition(true, nil)
+	conditions = append(conditions, vgReadinessCondition)
+
+	return conditions
+}
+
+func computeResourcesSyncedCondition(init bool, reconciliationError error) lvmv1alpha1.Condition {
+	condition := lvmv1alpha1.Condition{
+		Type: lvmv1alpha1.ResourcesSynced,
+	}
+	if init {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = ReasonReconciliationInProgress
+		condition.Message = MessageReconciliationInProgress
+		return condition
+	}
+
+	if reconciliationError == nil {
+		condition.Status = corev1.ConditionTrue
+		condition.Reason = ReasonResourcesSynced
+		condition.Message = MessageResourcesSynced
+	} else {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = ReasonResourcesSyncFailed
+		condition.Message = fmt.Sprintf(MessageReasonResourcesSyncFailed, reconciliationError)
+	}
+
+	return condition
+}
+
+func computeVGReadinessCondition(init bool, vgNodeStatusList *lvmv1alpha1.LVMVolumeGroupNodeStatusList) lvmv1alpha1.Condition {
+	condition := lvmv1alpha1.Condition{
+		Type: lvmv1alpha1.VolumeGroupsReady,
+	}
+	if init {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = ReasonVGReadinessInProgress
+		condition.Message = MessageVGReadinessInProgress
+		return condition
+	}
+
+	var isReady, isDegraded, isFailed, isProgressing bool
```

Comment:
I think this is quite a complicated logic, I think we could just write a small function that translates VGStatus => Reason+Message in a function with a switch case